pangeo_forge_recipes/storage.py

```diff
@@ -39,7 +43,14 @@ def _copy_btw_filesystems(input_opener, output_opener, BLOCK_SIZE=10_000_000):
                     ) from e
                 if not data:
                     break
-                logger.debug(f"_copy_btw_filesystems copying block of {len(data)} bytes")
+                # strided logging pattern for streaming transfers to avoid excessive logs
+                if streaming:
+                    summed_bytes += len(data)
+                    if summed_bytes // BLOCK_SIZE >= count:
+                        logger.debug(f"copying block {count} of ~{BLOCK_SIZE} bytes")
```

Comment:
Would be useful if we could say `copying block {count}/{total}`

But more fundamentally, I don't see how this logic actually reduces the frequency of output. Isn't `summed_bytes // BLOCK_SIZE >= count` always true, since we read data in in `BLOCK_SIZE` increments?

Reply:
> Would be useful if we could say copying block {count}/{total}

Yes this would be ideal, but in the case of streaming reads, we may not know the total size of the source file. This is the case for the motivating recipe https://github.com/pangeo-forge/staged-recipes/pull/56, where I've switched to streaming reads not because the HTTP server doesn't support range requests, but rather because the Content Length headers are incorrect, so range request reads terminate at the wrong place in the file.

> Isn't summed_bytes // BLOCK_SIZE >= count always true, since we read data in in BLOCK_SIZE increments?

As mentioned above, in the case of streaming reads, `data = source.read(BLOCK_SIZE)` actually ignores `BLOCK_SIZE` in favor of many smaller streaming reads. So perhaps this would be clearer as:
```python
if summed_bytes // LOGGING_INTERVAL >= count:
```
which is actually the behavior it exhibits when run.